Purchases/Networking/Backend.swift

```diff
@@ -749,33 +749,25 @@ private extension Backend {
 
     func getOfferingsCallbacksAndClearCache(forKey key: String) -> [OfferingsResponseHandler] {
         return callbackQueue.sync { [self] in
-            let callbacks = offeringsCallbacksCache.removeValue(forKey: key)
-            assert(callbacks != nil)
-            return callbacks ?? []
+            return offeringsCallbacksCache.removeValue(forKey: key)!
```

Comment:
`precondition` is nice because you can add a message to include any extra context around why this is going to crash, like size of the cache, etc. If you don't think there's any value in extra context here than this is fine with me 😄 , but I didn't write the initial implementation so I'll wait for somebody who has more knowledge on it to comment.

Reply:
That makes sense, I didn't know if there was any useful context to add here.

Reply:
RE [your comment](https://github.com/RevenueCat/purchases-ios/pull/1061#issuecomment-993061913) about irrecoverable errors: I don't have the context here, but maybe this isn't worth changing to a crash after all?

Reply:
Yeah, getOfferings should be idempotent, and anything relying on it can re-fetch if needed. I don't think this one needs a crash. Having a crash during dev could be helpful to find bugs, so I don't mind the `assert`. 

There is a larger conversation to be had, and an audit we should do to figure out the impact of not crashing vs crashing. I don't know what is correct, so I'd prefer to leave it functioning as is until we can do that analysis and apply our findings across all APIs.

Reply:
>  Having a crash during dev could be helpful to find bugs, so I don't mind the assert.

👍 

I'm on the same page, closing this.